### PR TITLE
Add enableRule switch into tidbmonitor.

### LINKED ISSUE #4110: Support add prometheus recoding rules when prometheus remote_write defined and alertmanager not defined
## Bug Report

**What version of Kubernetes are you using?**

**What version of TiDB Operator are you using?**

v1.2.0
**What storage classes exist in the Kubernetes cluster and what are used for PD/TiKV pods?**

**What's the status of the TiDB cluster pods?**

**What did you do?**

**What did you expect to see?**
Use  prometheus remote_write with recoding rules but without alertmanager. (Now We can custom recording rules by rebuild container tidb-monitor-initializer.)

**What did you see instead?**

When using prometheus remote_write and want to use recoding rules without alertmanager, the config of `RuleFiles` can't write to config file. Ref:
https://github.com/pingcap/tidb-operator/blob/33bf21cb626a8b8dc2dd586f8cd6d6c6875c448c/pkg/monitor/monitor/template.go#L467-L474

### PR DESCRIPTION
**Pull request by mikechengwei**
### What problem does this PR solve?
Fix #4110 
### What is changed and how does it work?
If not configuring AlertManager , it can add rule configuration through this switch.

### Code changes

- [ ] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 

### Release Notes

```release-note
Add enableRules switch into tidbmonitor. If not configuring AlertManager , it can add rule configuration through this switch.
```

### CODE REVIEW

File: pkg/monitor/monitor/template.go [hunk 1]
```diff
@@ -471,7 +472,15 @@ func RenderPrometheusConfig(model *MonitorConfigModel) (string, error) {
 		pc.RuleFiles = []string{
 			"/prometheus-rules/rules/*.rules.yml",
 		}
+	} else {
+		// If not configuring AlertManager,it also need to add rules when enabling `EnableRules` switch.
+		if model.EnableRules {
+			pc.RuleFiles = []string{
+				"/prometheus-rules/rules/*.rules.yml",
+			}
+		}
```
Comment (DanielZhangQD) [inline pkg/monitor/monitor/template.go:481]:
```suggestion
	} else if model.EnableAlertRules {
		// Add alert rules when `EnableAlertRules` enabled even if AlertManager not configured.
		pc.RuleFiles = []string{
			"/prometheus-rules/rules/*.rules.yml",
		}
```

**Comment by DanielZhangQD** [comment]
@mikechengwei Please fix the CI.

**Review by DanielZhangQD — APPROVED** [review]
LGTM

File: pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go [hunk 2]
```diff
@@ -121,6 +121,11 @@ type TidbMonitorSpec struct {
 	// PodSecurityContext of the component
 	// +optional
 	PodSecurityContext *corev1.PodSecurityContext `json:"podSecurityContext,omitempty"`
+
+	// EnableAlertRules adds alert rules to the Prometheus config even
+	// if If `AlertmanagerURL` is not configured.
```
Comment (csuzhangxc) [inline pkg/apis/pingcap/v1alpha1/tidbmonitor_types.go:126]:
```suggestion
	// if `AlertmanagerURL` is not configured.
```

**Review by csuzhangxc — APPROVED** [review]

**Pull request merged**